# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserParent.h
# Commit: a902acfbc868
# Full Hash: a902acfbc8680811e4538f8d27e78e4fe6ca55f2
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2019-05-23 15:55:11
# Regressor Bug: 1525720
# File Overlap Count: 1
# Description:
#   Bug 1525720, part 13 - Stop inheriting nsIRemoteTab interface in BrowserParent. r=nika
#   
#   This commit removes nsIRemoteTab as a parent class from BrowserParent,
#   so that BrowserHost is the only concrete implementation of nsIRemoteTab.
#   
# ==============================================================================

diff -r 50475885250c -r a902acfbc868 dom/ipc/BrowserParent.h
--- a/dom/ipc/BrowserParent.h	Wed May 08 11:45:23 2019 -0500
+++ b/dom/ipc/BrowserParent.h	Wed May 08 14:34:47 2019 -0500
@@ -86,7 +86,6 @@
  */
 class BrowserParent final : public PBrowserParent,
                             public nsIDOMEventListener,
-                            public nsIRemoteTab,
                             public nsIAuthPromptProvider,
                             public nsIKeyEventInPluginCallback,
                             public nsSupportsWeakReference,
@@ -104,12 +103,10 @@
 
   NS_DECL_CYCLE_COLLECTING_ISUPPORTS
   NS_DECL_NSIAUTHPROMPTPROVIDER
-  // nsIRemoteTab
-  NS_DECL_NSIREMOTETAB
   // nsIDOMEventListener interfaces
   NS_DECL_NSIDOMEVENTLISTENER
 
-  NS_DECL_CYCLE_COLLECTION_CLASS_AMBIGUOUS(BrowserParent, nsIRemoteTab)
+  NS_DECL_CYCLE_COLLECTION_CLASS_AMBIGUOUS(BrowserParent, nsIDOMEventListener)
 
   BrowserParent(ContentParent* aManager, const TabId& aTabId,
                 const TabContext& aContext,
@@ -127,8 +124,6 @@
 
   static BrowserParent* GetFrom(nsFrameLoader* aFrameLoader);
 
-  static BrowserParent* GetFrom(nsIRemoteTab* aBrowserParent);
-
   static BrowserParent* GetFrom(PBrowserParent* aBrowserParent);
 
   static BrowserParent* GetFrom(nsIContent* aContent);
@@ -678,6 +673,35 @@
 
   void SkipBrowsingContextDetach();
 
+  NS_IMETHOD GetDocShellIsActive(bool* aDocShellIsActive);
+  NS_IMETHOD SetDocShellIsActive(bool aDocShellIsActive);
+  NS_IMETHOD GetRenderLayers(bool* aRenderLayers);
+  NS_IMETHOD SetRenderLayers(bool aRenderLayers);
+  NS_IMETHOD GetHasLayers(bool* aHasLayers);
+  NS_IMETHOD ForceRepaint(void);
+  NS_IMETHOD NotifyResolutionChanged(void);
+  NS_IMETHOD Deprioritize(void);
+  NS_IMETHOD PreserveLayers(bool aPreserveLayers);
+  NS_IMETHOD GetTabId(uint64_t* aTabId);
+  NS_IMETHOD GetContentProcessId(uint64_t* aContentProcessId);
+  NS_IMETHOD GetOsPid(int32_t* aOsPid);
+  NS_IMETHOD GetHasContentOpener(bool* aHasContentOpener);
+  NS_IMETHOD GetHasPresented(bool* aHasPresented);
+  NS_IMETHOD GetWindowGlobalParents(
+      nsTArray<RefPtr<WindowGlobalParent>>& aWindowGlobalParents);
+  NS_IMETHOD TransmitPermissionsForPrincipal(nsIPrincipal* aPrincipal);
+  NS_IMETHOD GetHasBeforeUnload(bool* aHasBeforeUnload);
+  NS_IMETHOD GetOwnerElement(mozilla::dom::Element** aOwnerElement);
+  NS_IMETHOD StartApzAutoscroll(float aAnchorX, float aAnchorY,
+                                nsViewID aScrollId, uint32_t aPresShellId,
+                                bool* _retval);
+  NS_IMETHOD StopApzAutoscroll(nsViewID aScrollId, uint32_t aPresShellId);
+  NS_IMETHOD SaveRecording(const nsAString& aFileName, bool* _retval);
+  NS_IMETHOD GetContentBlockingLog(::mozilla::dom::Promise** _retval);
+  NS_IMETHOD MaybeCancelContentJSExecutionFromScript(
+      nsIRemoteTab::NavigationType aNavigationType,
+      JS::Handle<JS::Value> aCancelContentJSOptions, JSContext* aCx);
+
  protected:
   friend BrowserBridgeParent;
   friend BrowserHost;